components/Themes/src/MDCSemanticColorScheme.m

```diff
@@ -0,0 +1,63 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCSemanticColorScheme.h"
+
+@implementation MDCSemanticColorScheme
+
+@synthesize primaryColor = _primaryColor;
+@synthesize primaryColorLightVariant = _primaryColorLightVariant;
+@synthesize primaryColorDarkVariant = _primaryColorDarkVariant;
+@synthesize secondaryColor = _secondaryColor;
+@synthesize errorColor = _errorColor;
+
+- (instancetype)initWithPrimaryColor:(UIColor *)primaryColor
+            primaryColorLightVariant:(UIColor *)primaryColorLightVariant
+             primaryColorDarkVariant:(UIColor *)primaryColorDarkVariant
+                      secondaryColor:(UIColor *)secondaryColor
+                          errorColor:(UIColor *)errorColor {
+  self = [super init];
+  if (self) {
+    _primaryColor = primaryColor;
+    _primaryColorDarkVariant = primaryColorDarkVariant;
+    _primaryColorLightVariant = primaryColorLightVariant;
+    if (errorColor) {
+      _errorColor = errorColor;
+    } else {
+      // Default is #ff1744
+      _errorColor = [[UIColor alloc] initWithRed:1
+                                           green:(float)(0x17 / 255.0)
```

Comment:
Super-nit: CGFloat here and below.

Reply:
In light of #2931 (UIColor and NSCoding), I intentionally went with floats here so that components using this color value won't have to do gnarly work-arounds to check color equality.  An alternative (thanks @ajsecord), might be to place a category on UIColor:

```objectivec
+ (BOOL)mdc_isApproximatelyEqualToColor:(UIColor *)otherColor;
```

Reply:
The underlying data type for UIColor is CGFloat though, so anyone consuming a UIColor wouldn't be able to benefit from any float casting here (they can't know that we did and will have to assume the type is CGFloat).

Reply:
Can you clarify what you mean, @jverkoey ? As far as I understand from Apple's docs, color data is represented internally as a sequence of either 8-bit or 16-bit values (depending on RGB or P3 color spaces).  As far as I can determine, the CGFloat values are only used for equality testing.

Reply:
We'll likely remove the default from the code, so this discussion can be punted til the next time we implement hex colors in code. At that point, the decision will either be:

1. Document why a (float) cast is being used instead of a CGFloat.
2. Cast to CGFloat.